File: olymp.tj/7-8_region_2026(round 2)/j.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long

int dist(int a, int b, int n) {
    int d = abs(a - b);
    return min(d, n - d);
}

int main() {
    int n;
    cin >> n;
    vector<int> balls;
    vector<int> emp;
    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;
        for (int j = 0; j < x; j++)
            balls.push_back(i);
        if (x == 0)
            emp.push_back(i);
    }
    int m = balls.size();
    if (m <= 1) {
        cout << 0;
        return 0;
    }
    ll ans = 1e18+1;
    for (int j = 0; j < m; j++) {
        ll cost = 0;
        for (int i = 0; i < m; i++) {
            int bp = balls[i];
            int ep = emp[(i + j) % m];
            cost += dist(bp, ep, n);
        }
        ans = min(ans, cost);
    }
    cout << ans;
}

File: olymp.tj/National Olympiads/7-8_region_2026(round 2)/j.py
def dist(a, b, n):
    d = abs(a - b)
    return min(d, n - d)

def main():
    n = int(input())
    lst = list(map(int, input().split()))
    balls = []
    emp = []
    for i in range(n):
        x = lst[i]
        for j in range(x):
            balls.append(i)
        if x == 0:
            emp.append(i)
    m = len(balls)
    if m <= 1:
        print(0)
        return
    ans = 10**18
    for j in range(m):
        k = 0
        for i in range(m):
            bp = balls[i]
            ep = emp[(i + j) % m]
            k += dist(bp, ep, n)
        ans = min(ans, k)
    print(ans)
if __name__ == "__main__":
    main()
